lnwallet/btcwallet/psbt.go

```diff
@@ -344,6 +378,34 @@ func signSegWitV1KeySpend(in *psbt.PInput, tx *wire.MsgTx,
 	return nil
 }
 
+// signSegWitV1ScriptSpend attempts to generate a signature for a SegWit version
+// 1 (p2tr) input and stores it in the TaprootScriptSpendSig field.
+func signSegWitV1ScriptSpend(in *psbt.PInput, tx *wire.MsgTx,
+	sigHashes *txscript.TxSigHashes, idx int, privKey *btcec.PrivateKey,
+	leaf txscript.TapLeaf) error {
+
+	rawSig, err := txscript.RawTxInTapscriptSignature(
+		tx, sigHashes, idx, in.WitnessUtxo.Value,
+		in.WitnessUtxo.PkScript, leaf, in.SighashType, privKey,
+	)
+	if err != nil {
+		return fmt.Errorf("error signing taproot script input %d: %v",
+			idx, err)
+	}
+
+	leafHash := leaf.TapHash()
+	in.TaprootScriptSpendSig = append(
+		in.TaprootScriptSpendSig, &psbt.TaprootScriptSpendSig{
+			XOnlyPubKey: in.TaprootBip32Derivation[0].XOnlyPubKey,
+			LeafHash:    leafHash[:],
```

Comment:
May be worth adding a comment here that this is snipping the sighash byte off the end of the sig.